<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>example • rms2</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="example">
<meta property="og:description" content="rms2">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rms2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/example.html">example</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/explodecomputer/rms2/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="example_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>example</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/explodecomputer/rms2/blob/master/vignettes/example.Rmd"><code>vignettes/example.Rmd</code></a></small>
      <div class="hidden name"><code>example.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/explodecomputer/rms2">rms2</a></span><span class="op">)</span>
<span class="co"># library(devtools)</span>
<span class="co"># load_all()</span></pre></div>
<p>Create new rms2 class for the outcome GWAS of interest</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="va">x</span> <span class="op">&lt;-</span> <span class="va"><a href="../reference/rms2.html">rms2</a></span><span class="op">$</span><span class="fu">new</span><span class="op">(</span><span class="st">"ukb-b-2227"</span><span class="op">)</span>
<span class="co">#&gt; API: public: http://gwas-api.mrcieu.ac.uk/</span>
<span class="co">#&gt; Classes 'GwasInfo' and 'data.frame': 1 obs. of  18 variables:</span>
<span class="co">#&gt;  $ id         : chr "ukb-b-2227"</span>
<span class="co">#&gt;  $ trait      : chr "Number of children fathered"</span>
<span class="co">#&gt;  $ sex        : chr "Males and Females"</span>
<span class="co">#&gt;  $ mr         : int 1</span>
<span class="co">#&gt;  $ subcategory: logi NA</span>
<span class="co">#&gt;  $ priority   : int 1</span>
<span class="co">#&gt;  $ note       : chr "2405: Output from GWAS pipeline using Phesant derived variables from UKBiobank"</span>
<span class="co">#&gt;  $ ontology   : logi NA</span>
<span class="co">#&gt;  $ year       : int 2018</span>
<span class="co">#&gt;  $ author     : chr "Ben Elsworth"</span>
<span class="co">#&gt;  $ unit       : chr "SD"</span>
<span class="co">#&gt;  $ group_name : chr "public"</span>
<span class="co">#&gt;  $ sample_size: int 209872</span>
<span class="co">#&gt;  $ population : chr "European"</span>
<span class="co">#&gt;  $ build      : chr "HG19/GRCh37"</span>
<span class="co">#&gt;  $ nsnp       : int 9851867</span>
<span class="co">#&gt;  $ consortium : chr "MRC-IEU"</span>
<span class="co">#&gt;  $ category   : chr "Categorical Ordered"</span>
<span class="co">#&gt; </span></pre></div>
<p>Get the GWAS hits for that dataset</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="va">x</span><span class="op">$</span><span class="fu">extract_gwashits</span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Found 3 GWAS hits for ukb-b-2227</span>
<span class="va">x</span><span class="op">$</span><span class="va">gwashits</span>
<span class="co">#&gt; # A tibble: 3 x 12</span>
<span class="co">#&gt;      beta chr        se        p      n  position id     rsid  ea    nea     eaf</span>
<span class="co">#&gt;     &lt;dbl&gt; &lt;chr&gt;   &lt;dbl&gt;    &lt;dbl&gt;  &lt;dbl&gt;     &lt;int&gt; &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;dbl&gt;</span>
<span class="co">#&gt; 1 -0.0193 3     0.00260 1.30e-13 209872  85510543 ukb-b… rs62… T     A     0.617</span>
<span class="co">#&gt; 2  0.0187 7     0.00275 1.10e-11 209872 114940159 ukb-b… rs69… A     G     0.694</span>
<span class="co">#&gt; 3 -0.0167 8     0.00302 2.80e- 8 209872  65133678 ukb-b… rs23… C     T     0.772</span>
<span class="co">#&gt; # … with 1 more variable: trait &lt;chr&gt;</span></pre></div>
<p>Scan OpenGWAS for associations with the second GWAS hit</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="va">x</span><span class="op">$</span><span class="fu">scan_rsid</span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">gwashits</span><span class="op">$</span><span class="va">rsid</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; Found 6 associations for rs6968380</span>
<span class="va">x</span><span class="op">$</span><span class="va">rsid_scan</span>
<span class="co">#&gt; $rs6968380</span>
<span class="co">#&gt; # A tibble: 6 x 12</span>
<span class="co">#&gt;    position        p      se     beta      n chr   id    rsid  ea    nea     eaf</span>
<span class="co">#&gt;       &lt;int&gt;    &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;dbl&gt;</span>
<span class="co">#&gt; 1 114940159 9.20e-18 8.18e-4 -0.00702 462933 7     ukb-… rs69… A     G     0.693</span>
<span class="co">#&gt; 2 114940159 6.50e-10 9.96e-4  0.00615 459600 7     ukb-… rs69… A     G     0.693</span>
<span class="co">#&gt; 3 114940159 1.30e- 9 1.24e-3 -0.00751 462434 7     ukb-… rs69… A     G     0.693</span>
<span class="co">#&gt; 4 114940159 3.90e- 9 1.07e-3 -0.00629 451893 7     ukb-… rs69… A     G     0.693</span>
<span class="co">#&gt; 5 114940159 4.82e- 9 3.19e-3  0.0187  154888 7     ukb-… rs69… A     G     0.695</span>
<span class="co">#&gt; 6 114940159 2.20e- 8 1.06e-3 -0.00594 459560 7     ukb-… rs69… A     G     0.693</span>
<span class="co">#&gt; # … with 1 more variable: trait &lt;chr&gt;</span></pre></div>
<p>Perform colocalisation for each of the candidate associations</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="va">x</span><span class="op">$</span><span class="fu">coloc_scan</span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">gwashits</span><span class="op">$</span><span class="va">rsid</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; coloc analysis of ukb-b-11497</span>
<span class="co">#&gt; Warning in sdY.est(d$varbeta, d$MAF, d$N): estimating sdY from maf and varbeta,</span>
<span class="co">#&gt; please directly supply sdY if known</span>

<span class="co">#&gt; Warning in sdY.est(d$varbeta, d$MAF, d$N): estimating sdY from maf and varbeta,</span>
<span class="co">#&gt; please directly supply sdY if known</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  2.37e-19  4.59e-14  5.81e-08  1.03e-02  9.90e-01 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 99%"</span>
<span class="co">#&gt; coloc analysis of ukb-b-698</span>
<span class="co">#&gt; Warning in sdY.est(d$varbeta, d$MAF, d$N): estimating sdY from maf and varbeta,</span>
<span class="co">#&gt; please directly supply sdY if known</span>

<span class="co">#&gt; Warning in sdY.est(d$varbeta, d$MAF, d$N): estimating sdY from maf and varbeta,</span>
<span class="co">#&gt; please directly supply sdY if known</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  2.03e-11  3.93e-06  6.61e-08  1.18e-02  9.88e-01 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 98.8%"</span>
<span class="co">#&gt; coloc analysis of ukb-b-223</span>
<span class="co">#&gt; Warning in sdY.est(d$varbeta, d$MAF, d$N): estimating sdY from maf and varbeta,</span>
<span class="co">#&gt; please directly supply sdY if known</span>

<span class="co">#&gt; Warning in sdY.est(d$varbeta, d$MAF, d$N): estimating sdY from maf and varbeta,</span>
<span class="co">#&gt; please directly supply sdY if known</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  2.84e-14  5.51e-09  5.01e-06  9.71e-01  2.92e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 2.92%"</span>
<span class="co">#&gt; coloc analysis of ukb-b-13764</span>
<span class="co">#&gt; Warning in sdY.est(d$varbeta, d$MAF, d$N): estimating sdY from maf and varbeta,</span>
<span class="co">#&gt; please directly supply sdY if known</span>

<span class="co">#&gt; Warning in sdY.est(d$varbeta, d$MAF, d$N): estimating sdY from maf and varbeta,</span>
<span class="co">#&gt; please directly supply sdY if known</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.52e-11  6.81e-06  7.09e-08  1.27e-02  9.87e-01 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 98.7%"</span>
<span class="co">#&gt; coloc analysis of ukb-a-304</span>
<span class="co">#&gt; Warning in sdY.est(d$varbeta, d$MAF, d$N): estimating sdY from maf and varbeta,</span>
<span class="co">#&gt; please directly supply sdY if known</span>

<span class="co">#&gt; Warning in sdY.est(d$varbeta, d$MAF, d$N): estimating sdY from maf and varbeta,</span>
<span class="co">#&gt; please directly supply sdY if known</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  1.27e-10  2.46e-05  5.68e-08  1.00e-02  9.90e-01 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 99%"</span>
<span class="co">#&gt; coloc analysis of ukb-b-6991</span>
<span class="co">#&gt; Warning in sdY.est(d$varbeta, d$MAF, d$N): estimating sdY from maf and varbeta,</span>
<span class="co">#&gt; please directly supply sdY if known</span>

<span class="co">#&gt; Warning in sdY.est(d$varbeta, d$MAF, d$N): estimating sdY from maf and varbeta,</span>
<span class="co">#&gt; please directly supply sdY if known</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  9.63e-10  1.87e-04  9.19e-08  1.68e-02  9.83e-01 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 98.3%"</span>
<span class="va">x</span><span class="op">$</span><span class="va">coloc_result</span>
<span class="co">#&gt; $rs6968380</span>
<span class="co">#&gt; # A tibble: 6 x 7</span>
<span class="co">#&gt;   id          nsnps PP.H0.abf PP.H1.abf    PP.H2.abf PP.H3.abf PP.H4.abf</span>
<span class="co">#&gt;   &lt;chr&gt;       &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;        &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;</span>
<span class="co">#&gt; 1 ukb-b-11497   297  2.37e-19  4.59e-14 0.0000000581    0.0103    0.990 </span>
<span class="co">#&gt; 2 ukb-b-698     325  2.03e-11  3.93e- 6 0.0000000661    0.0118    0.988 </span>
<span class="co">#&gt; 3 ukb-b-223     324  2.84e-14  5.51e- 9 0.00000501      0.971     0.0292</span>
<span class="co">#&gt; 4 ukb-b-13764   315  3.52e-11  6.81e- 6 0.0000000709    0.0127    0.987 </span>
<span class="co">#&gt; 5 ukb-a-304     325  1.27e-10  2.46e- 5 0.0000000568    0.0100    0.990 </span>
<span class="co">#&gt; 6 ukb-b-6991    318  9.63e-10  1.87e- 4 0.0000000919    0.0168    0.983</span></pre></div>
<p>Example of plotting a region (extracts LD matrix)</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="va">x</span><span class="op">$</span><span class="fu">plot_coloc</span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">gwashits</span><span class="op">$</span><span class="va">rsid</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, <span class="va">x</span><span class="op">$</span><span class="va">rsid_scan</span><span class="op">[[</span><span class="va">x</span><span class="op">$</span><span class="va">gwashits</span><span class="op">$</span><span class="va">rsid</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; Extracting LD matrix for 297 variants</span>
<span class="co">#&gt; Please look at vignettes for options on running this locally if you need to run many instances of this command.</span>
<span class="co">#&gt; Warning in ieugwasr::ld_matrix(markers[["marker"]], with_alleles = FALSE, : The following variants are not present in the LD reference panel</span>
<span class="co">#&gt; rs117161781</span>
<span class="co">#&gt; rs117273777</span>
<span class="co">#&gt; rs117303422</span>
<span class="co">#&gt; rs118169575</span>
<span class="co">#&gt; rs139454624</span>
<span class="co">#&gt; rs139462051</span>
<span class="co">#&gt; rs139500510</span>
<span class="co">#&gt; rs140651620</span>
<span class="co">#&gt; rs141744989</span>
<span class="co">#&gt; rs141995076</span>
<span class="co">#&gt; rs142087962</span>
<span class="co">#&gt; rs142329760</span>
<span class="co">#&gt; rs142372229</span>
<span class="co">#&gt; rs142664107</span>
<span class="co">#&gt; rs143271578</span>
<span class="co">#&gt; rs144588831</span>
<span class="co">#&gt; rs146170513</span>
<span class="co">#&gt; rs151172702</span>
<span class="co">#&gt; rs180940258</span>
<span class="co">#&gt; rs180946512</span>
<span class="co">#&gt; rs181430058</span>
<span class="co">#&gt; rs181494117</span>
<span class="co">#&gt; rs182895259</span>
<span class="co">#&gt; rs183744920</span>
<span class="co">#&gt; rs184050099</span>
<span class="co">#&gt; rs184314722</span>
<span class="co">#&gt; rs185309780</span>
<span class="co">#&gt; rs187219455</span>
<span class="co">#&gt; rs193077931</span>
<span class="co">#&gt; rs2188543</span>
<span class="co">#&gt; rs2396813</span>
<span class="co">#&gt; rs4730672</span>
<span class="co">#&gt; rs558482618</span>
<span class="co">#&gt; rs560974722</span>
<span class="co">#&gt; rs575755508</span>
<span class="co">#&gt; rs71563623</span>
<span class="co">#&gt; rs73200222</span>
<span class="co">#&gt; rs75351137</span>
<span class="co">#&gt; rs75354202</span>
<span class="co">#&gt; rs77870086</span>
<span class="co">#&gt; rs7787817</span>
<span class="co">#&gt; Found 256 variants in LD reference panel</span>
<span class="co">#&gt; Loading required package: gassocplot</span></pre></div>
<p><img src="example_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<pre><code>#&gt; TableGrob (2 x 1) "arrange": 2 grobs
#&gt;   z     cells    name              grob
#&gt; 1 1 (1-1,1-1) arrange    gtable[layout]
#&gt; 2 2 (2-2,1-1) arrange gtable[guide-box]</code></pre>
<p>Perform MR for all the candidate traits (though could also exclude the traits that don’t colocalise)</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="va">x</span><span class="op">$</span><span class="fu">mr</span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">gwashits</span><span class="op">$</span><span class="va">rsid</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, exclude_rsid_region<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; Extracting data for 79 SNP(s) from 1 GWAS(s)</span>
<span class="co">#&gt; Harmonising Operation code: tonsillectomy +/- adenoids || id:ukb-b-11497 (ukb-b-11497) and Number of children fathered || id:ukb-b-2227 (ukb-b-2227)</span>
<span class="co">#&gt; Removing the following SNPs for being palindromic with intermediate allele frequencies:</span>
<span class="co">#&gt; rs1056688, rs2215552, rs2661798</span>
<span class="co">#&gt; [1] 79</span>
<span class="co">#&gt; Estimating correlation for quantitative trait.</span>
<span class="co">#&gt; This method is an approximation, and may be numerically unstable.</span>
<span class="co">#&gt; Ideally you should estimate r directly from independent replication samples.</span>
<span class="co">#&gt; Use get_r_from_lor for binary traits.</span>
<span class="co">#&gt; Estimating correlation for quantitative trait.</span>
<span class="co">#&gt; This method is an approximation, and may be numerically unstable.</span>
<span class="co">#&gt; Ideally you should estimate r directly from independent replication samples.</span>
<span class="co">#&gt; Use get_r_from_lor for binary traits.</span>
<span class="co">#&gt; [1] 78</span>
<span class="co">#&gt; [1] 78</span>
<span class="co">#&gt; Analysing 'ukb-b-11497' on 'ukb-b-2227'</span>
<span class="co">#&gt; Extracting data for 5 SNP(s) from 1 GWAS(s)</span>
<span class="co">#&gt; Harmonising How are people in household related to participant: Husband, wife or partner || id:ukb-b-698 (ukb-b-698) and Number of children fathered || id:ukb-b-2227 (ukb-b-2227)</span>
<span class="co">#&gt; [1] 5</span>
<span class="co">#&gt; Estimating correlation for quantitative trait.</span>
<span class="co">#&gt; This method is an approximation, and may be numerically unstable.</span>
<span class="co">#&gt; Ideally you should estimate r directly from independent replication samples.</span>
<span class="co">#&gt; Use get_r_from_lor for binary traits.</span>
<span class="co">#&gt; Estimating correlation for quantitative trait.</span>
<span class="co">#&gt; This method is an approximation, and may be numerically unstable.</span>
<span class="co">#&gt; Ideally you should estimate r directly from independent replication samples.</span>
<span class="co">#&gt; Use get_r_from_lor for binary traits.</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; Analysing 'ukb-b-698' on 'ukb-b-2227'</span>
<span class="co">#&gt; Extracting data for 35 SNP(s) from 1 GWAS(s)</span>
<span class="co">#&gt; Harmonising Current tobacco smoking || id:ukb-b-223 (ukb-b-223) and Number of children fathered || id:ukb-b-2227 (ukb-b-2227)</span>
<span class="co">#&gt; Removing the following SNPs for being palindromic with intermediate allele frequencies:</span>
<span class="co">#&gt; rs58040999</span>
<span class="co">#&gt; [1] 35</span>
<span class="co">#&gt; Estimating correlation for quantitative trait.</span>
<span class="co">#&gt; This method is an approximation, and may be numerically unstable.</span>
<span class="co">#&gt; Ideally you should estimate r directly from independent replication samples.</span>
<span class="co">#&gt; Use get_r_from_lor for binary traits.</span>
<span class="co">#&gt; Estimating correlation for quantitative trait.</span>
<span class="co">#&gt; This method is an approximation, and may be numerically unstable.</span>
<span class="co">#&gt; Ideally you should estimate r directly from independent replication samples.</span>
<span class="co">#&gt; Use get_r_from_lor for binary traits.</span>
<span class="co">#&gt; [1] 35</span>
<span class="co">#&gt; [1] 34</span>
<span class="co">#&gt; Analysing 'ukb-b-223' on 'ukb-b-2227'</span>
<span class="co">#&gt; Extracting data for 29 SNP(s) from 1 GWAS(s)</span>
<span class="co">#&gt; Harmonising Long-standing illness, disability or infirmity || id:ukb-b-13764 (ukb-b-13764) and Number of children fathered || id:ukb-b-2227 (ukb-b-2227)</span>
<span class="co">#&gt; [1] 30</span>
<span class="co">#&gt; Estimating correlation for quantitative trait.</span>
<span class="co">#&gt; This method is an approximation, and may be numerically unstable.</span>
<span class="co">#&gt; Ideally you should estimate r directly from independent replication samples.</span>
<span class="co">#&gt; Use get_r_from_lor for binary traits.</span>
<span class="co">#&gt; Estimating correlation for quantitative trait.</span>
<span class="co">#&gt; This method is an approximation, and may be numerically unstable.</span>
<span class="co">#&gt; Ideally you should estimate r directly from independent replication samples.</span>
<span class="co">#&gt; Use get_r_from_lor for binary traits.</span>
<span class="co">#&gt; [1] 28</span>
<span class="co">#&gt; [1] 27</span>
<span class="co">#&gt; Analysing 'ukb-b-13764' on 'ukb-b-2227'</span>
<span class="co">#&gt; Extracting data for 2 SNP(s) from 1 GWAS(s)</span>
<span class="co">#&gt; Harmonising Number of children fathered || id:ukb-a-304 (ukb-a-304) and Number of children fathered || id:ukb-b-2227 (ukb-b-2227)</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; Estimating correlation for quantitative trait.</span>
<span class="co">#&gt; This method is an approximation, and may be numerically unstable.</span>
<span class="co">#&gt; Ideally you should estimate r directly from independent replication samples.</span>
<span class="co">#&gt; Use get_r_from_lor for binary traits.</span>
<span class="co">#&gt; Estimating correlation for quantitative trait.</span>
<span class="co">#&gt; This method is an approximation, and may be numerically unstable.</span>
<span class="co">#&gt; Ideally you should estimate r directly from independent replication samples.</span>
<span class="co">#&gt; Use get_r_from_lor for binary traits.</span>
<span class="co">#&gt; [1] 2</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; Analysing 'ukb-a-304' on 'ukb-b-2227'</span>
<span class="co">#&gt; Extracting data for 44 SNP(s) from 1 GWAS(s)</span>
<span class="co">#&gt; Harmonising Seen doctor (GP) for nerves, anxiety, tension or depression || id:ukb-b-6991 (ukb-b-6991) and Number of children fathered || id:ukb-b-2227 (ukb-b-2227)</span>
<span class="co">#&gt; Removing the following SNPs for incompatible alleles:</span>
<span class="co">#&gt; rs12159707, rs12159707, rs34555420, rs34555420</span>
<span class="co">#&gt; Removing the following SNPs for being palindromic with intermediate allele frequencies:</span>
<span class="co">#&gt; rs10143492, rs2876520, rs393488</span>
<span class="co">#&gt; [1] 50</span>
<span class="co">#&gt; Estimating correlation for quantitative trait.</span>
<span class="co">#&gt; This method is an approximation, and may be numerically unstable.</span>
<span class="co">#&gt; Ideally you should estimate r directly from independent replication samples.</span>
<span class="co">#&gt; Use get_r_from_lor for binary traits.</span>
<span class="co">#&gt; Estimating correlation for quantitative trait.</span>
<span class="co">#&gt; This method is an approximation, and may be numerically unstable.</span>
<span class="co">#&gt; Ideally you should estimate r directly from independent replication samples.</span>
<span class="co">#&gt; Use get_r_from_lor for binary traits.</span>
<span class="co">#&gt; [1] 49</span>
<span class="co">#&gt; [1] 48</span>
<span class="co">#&gt; Analysing 'ukb-b-6991' on 'ukb-b-2227'</span>
<span class="va">x</span><span class="op">$</span><span class="va">mr_scan</span><span class="op">[[</span><span class="va">x</span><span class="op">$</span><span class="va">gwashits</span><span class="op">$</span><span class="va">rsid</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">exposure</span>, <span class="va">b</span>, <span class="va">se</span>, <span class="va">pval</span><span class="op">)</span>
<span class="co">#&gt;                                                                                       exposure</span>
<span class="co">#&gt; 1                                 Operation code: tonsillectomy +/- adenoids || id:ukb-b-11497</span>
<span class="co">#&gt; 2 How are people in household related to participant: Husband, wife or partner || id:ukb-b-698</span>
<span class="co">#&gt; 3                                                      Current tobacco smoking || id:ukb-b-223</span>
<span class="co">#&gt; 4                             Long-standing illness, disability or infirmity || id:ukb-b-13764</span>
<span class="co">#&gt; 5                                                  Number of children fathered || id:ukb-a-304</span>
<span class="co">#&gt; 6                 Seen doctor (GP) for nerves, anxiety, tension or depression || id:ukb-b-6991</span>
<span class="co">#&gt;              b         se         pval</span>
<span class="co">#&gt; 1 -0.009752799 0.05883241 8.683360e-01</span>
<span class="co">#&gt; 2  1.600228195 0.34060979 2.625577e-06</span>
<span class="co">#&gt; 3  0.119799051 0.05984600 4.530744e-02</span>
<span class="co">#&gt; 4  0.113984410 0.09027558 2.067232e-01</span>
<span class="co">#&gt; 5  0.967971548 0.13132069 1.692875e-13</span>
<span class="co">#&gt; 6  0.058641007 0.08016851 4.644910e-01</span></pre></div>
<p>Perform multivariable MR of candidate traits. Excluding the alternative GWAS of number of children fathered as this is synonymous with the outcome</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="va">x</span><span class="op">$</span><span class="fu">mvmr</span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">gwashits</span><span class="op">$</span><span class="va">rsid</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, traitlist<span class="op">=</span><span class="va">x</span><span class="op">$</span><span class="va">rsid_scan</span><span class="op">[[</span><span class="va">x</span><span class="op">$</span><span class="va">gwashits</span><span class="op">$</span><span class="va">rsid</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="op">-</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; Please look at vignettes for options on running this locally if you need to run many instances of this command.</span>
<span class="co">#&gt; Clumping 1, 192 variants, using EUR population reference</span>
<span class="co">#&gt; Removing 51 of 192 variants due to LD with other variants or absence from LD reference panel</span>
<span class="co">#&gt; Extracting data for 141 SNP(s) from 5 GWAS(s)</span>
<span class="co">#&gt; Harmonising Operation code: tonsillectomy +/- adenoids || id:ukb-b-11497 (ukb-b-11497) and Long-standing illness, disability or infirmity || id:ukb-b-13764 (ukb-b-13764)</span>
<span class="co">#&gt; Removing the following SNPs for being palindromic with intermediate allele frequencies:</span>
<span class="co">#&gt; rs10143492, rs1056688, rs2215552, rs2661798, rs58040999</span>
<span class="co">#&gt; Harmonising Operation code: tonsillectomy +/- adenoids || id:ukb-b-11497 (ukb-b-11497) and Current tobacco smoking || id:ukb-b-223 (ukb-b-223)</span>
<span class="co">#&gt; Removing the following SNPs for being palindromic with intermediate allele frequencies:</span>
<span class="co">#&gt; rs10143492, rs1056688, rs2215552, rs2661798, rs58040999</span>
<span class="co">#&gt; Harmonising Operation code: tonsillectomy +/- adenoids || id:ukb-b-11497 (ukb-b-11497) and How are people in household related to participant: Husband, wife or partner || id:ukb-b-698 (ukb-b-698)</span>
<span class="co">#&gt; Removing the following SNPs for being palindromic with intermediate allele frequencies:</span>
<span class="co">#&gt; rs10143492, rs1056688, rs2215552, rs2661798, rs58040999</span>
<span class="co">#&gt; Harmonising Operation code: tonsillectomy +/- adenoids || id:ukb-b-11497 (ukb-b-11497) and Seen doctor (GP) for nerves, anxiety, tension or depression || id:ukb-b-6991 (ukb-b-6991)</span>
<span class="co">#&gt; Removing the following SNPs for being palindromic with intermediate allele frequencies:</span>
<span class="co">#&gt; rs10143492, rs1056688, rs2215552, rs2661798, rs58040999</span>
<span class="co">#&gt; Extracting data for 141 SNP(s) from 1 GWAS(s)</span>
<span class="co">#&gt; Harmonising Operation code: tonsillectomy +/- adenoids || id:ukb-b-11497 (ukb-b-11497) and Number of children fathered || id:ukb-b-2227 (ukb-b-2227)</span>
<span class="co">#&gt; Removing the following SNPs for being palindromic with intermediate allele frequencies:</span>
<span class="co">#&gt; rs10143492, rs1056688, rs2215552, rs2661798, rs58040999</span>
<span class="va">x</span><span class="op">$</span><span class="va">mv_dat</span><span class="op">[[</span><span class="va">x</span><span class="op">$</span><span class="va">gwashits</span><span class="op">$</span><span class="va">rsid</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"result"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">exposure</span>, <span class="va">b</span>, <span class="va">se</span>, <span class="va">pval</span><span class="op">)</span>
<span class="co">#&gt;                                                                                       exposure</span>
<span class="co">#&gt; 1                                 Operation code: tonsillectomy +/- adenoids || id:ukb-b-11497</span>
<span class="co">#&gt; 2                             Long-standing illness, disability or infirmity || id:ukb-b-13764</span>
<span class="co">#&gt; 3                                                      Current tobacco smoking || id:ukb-b-223</span>
<span class="co">#&gt; 4 How are people in household related to participant: Husband, wife or partner || id:ukb-b-698</span>
<span class="co">#&gt; 5                 Seen doctor (GP) for nerves, anxiety, tension or depression || id:ukb-b-6991</span>
<span class="co">#&gt;              b         se         pval</span>
<span class="co">#&gt; 1 -0.003478738 0.06497898 9.573046e-01</span>
<span class="co">#&gt; 2  0.093665403 0.09138850 3.054035e-01</span>
<span class="co">#&gt; 3  0.222608807 0.08616459 9.779497e-03</span>
<span class="co">#&gt; 4  1.457358841 0.18169451 1.049481e-15</span>
<span class="co">#&gt; 5  0.123356060 0.10108759 2.223554e-01</span></pre></div>
<p>Perform feature selection and re-estimate associations after that. It updates the result from the standard MVMR analysis</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="va">x</span><span class="op">$</span><span class="fu">mvmr_lasso</span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">gwashits</span><span class="op">$</span><span class="va">rsid</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; Performing feature selection</span>
<span class="co">#&gt; Please look at vignettes for options on running this locally if you need to run many instances of this command.</span>
<span class="co">#&gt; Clumping 1, 114 variants, using EUR population reference</span>
<span class="co">#&gt; Removing 32 of 114 variants due to LD with other variants or absence from LD reference panel</span>
<span class="co">#&gt; Extracting data for 82 SNP(s) from 4 GWAS(s)</span>
<span class="co">#&gt; Harmonising Long-standing illness, disability or infirmity || id:ukb-b-13764 (ukb-b-13764) and Current tobacco smoking || id:ukb-b-223 (ukb-b-223)</span>
<span class="co">#&gt; Removing the following SNPs for being palindromic with intermediate allele frequencies:</span>
<span class="co">#&gt; rs10143492, rs2876520, rs58040999</span>
<span class="co">#&gt; Harmonising Long-standing illness, disability or infirmity || id:ukb-b-13764 (ukb-b-13764) and How are people in household related to participant: Husband, wife or partner || id:ukb-b-698 (ukb-b-698)</span>
<span class="co">#&gt; Removing the following SNPs for being palindromic with intermediate allele frequencies:</span>
<span class="co">#&gt; rs10143492, rs2876520, rs58040999</span>
<span class="co">#&gt; Harmonising Long-standing illness, disability or infirmity || id:ukb-b-13764 (ukb-b-13764) and Seen doctor (GP) for nerves, anxiety, tension or depression || id:ukb-b-6991 (ukb-b-6991)</span>
<span class="co">#&gt; Removing the following SNPs for being palindromic with intermediate allele frequencies:</span>
<span class="co">#&gt; rs10143492, rs2876520, rs58040999</span>
<span class="co">#&gt; Extracting data for 82 SNP(s) from 1 GWAS(s)</span>
<span class="co">#&gt; Harmonising Long-standing illness, disability or infirmity || id:ukb-b-13764 (ukb-b-13764) and Number of children fathered || id:ukb-b-2227 (ukb-b-2227)</span>
<span class="co">#&gt; Removing the following SNPs for being palindromic with intermediate allele frequencies:</span>
<span class="co">#&gt; rs10143492, rs2876520, rs58040999</span>
<span class="va">x</span><span class="op">$</span><span class="va">mv_dat</span><span class="op">[[</span><span class="va">x</span><span class="op">$</span><span class="va">gwashits</span><span class="op">$</span><span class="va">rsid</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"result"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">exposure</span>, <span class="va">b</span>, <span class="va">se</span>, <span class="va">pval</span><span class="op">)</span>
<span class="co">#&gt;                                                                                       exposure</span>
<span class="co">#&gt; 1                             Long-standing illness, disability or infirmity || id:ukb-b-13764</span>
<span class="co">#&gt; 2                                                      Current tobacco smoking || id:ukb-b-223</span>
<span class="co">#&gt; 3 How are people in household related to participant: Husband, wife or partner || id:ukb-b-698</span>
<span class="co">#&gt; 4                 Seen doctor (GP) for nerves, anxiety, tension or depression || id:ukb-b-6991</span>
<span class="co">#&gt;            b         se         pval</span>
<span class="co">#&gt; 1 0.07153219 0.09862285 4.682616e-01</span>
<span class="co">#&gt; 2 0.20287671 0.08644674 1.893336e-02</span>
<span class="co">#&gt; 3 1.52946203 0.20730583 1.609203e-13</span>
<span class="co">#&gt; 4 0.14516472 0.10670644 1.736996e-01</span></pre></div>
<p>Note that this might lead to under-estimated standard errors so it should be used for estimation rather than hypothesis testing</p>
<div id="todo" class="section level2">
<h2 class="hasAnchor">
<a href="#todo" class="anchor"></a>Todo</h2>
<ul>
<li>MVMR should not be performed for traits that don’t have genome-wide coverage because it will cause most SNPs to be excluded e.g. if one of the traits has less than 90% of the GWAS hits from the other traits then exclude it</li>
<li>Estimation of variance in GWAS hit mediated by candidate traits</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Gibran Hemani.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
