<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Reverse 2-sample MR analysis — rms2 • rms2</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Reverse 2-sample MR analysis — rms2" />
<meta property="og:description" content="This analysis is a wrapper around MR and colocalisation functions for systematically searching OpenGWAS for mediating pathways from genotype to trait" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rms2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/example.html">example</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/explodecomputer/rms2/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Reverse 2-sample MR analysis</h1>
    <small class="dont-index">Source: <a href='https://github.com/explodecomputer/rms2/blob/master/R/rms2.r'><code>R/rms2.r</code></a></small>
    <div class="hidden name"><code>rms2.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This analysis is a wrapper around MR and colocalisation functions for systematically searching OpenGWAS for mediating pathways from genotype to trait</p>
    </div>



    <h2 class="hasAnchor" id="public-fields"><a class="anchor" href="#public-fields"></a>Public fields</h2>

    <p><div class="r6-fields"></p><dl>
<dt><code>igd_id</code></dt><dd><p>The OpenGWAS dataset to analyse as the outcome GWAS</p></dd>

<dt><code>info</code></dt><dd><p>Meta data for igd_id</p></dd>

<dt><code>gwashits</code></dt><dd><p>Data from of GWAS hits</p></dd>

<dt><code>rsid_scan</code></dt><dd><p>List of results of PheWAS search - one item for each GWAS hit</p></dd>

<dt><code>coloc_data</code></dt><dd><p>Data used for colocalisation analysis - list of lists (rsid x candidate trait)</p></dd>

<dt><code>coloc_result</code></dt><dd><p>Results from colocalisation analysis - list of data frames, one data frame for each region analysed</p></dd>

<dt><code>candidate_instruments</code></dt><dd><p>List of instrument datasets for each candidate</p></dd>

<dt><code>candidate_instruments_outcome</code></dt><dd><p>Dataframe of instrument-outcome associations</p></dd>

<dt><code>mr_scan</code></dt><dd><p>Results from MR analysis - list of data frames with one data frame for each region</p></dd>

<dt><code>mv_dat</code></dt><dd><p>Data and results from mvmr</p></dd>

</dl><p></div></p>
    <h2 class="hasAnchor" id="methods"><a class="anchor" href="#methods"></a>Methods</h2>

    
<h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Public methods</h3>

<ul>
<li><p><a href='#method-new'><code>rms2$new()</code></a></p></li>
<li><p><a href='#method-extract_gwashits'><code>rms2$extract_gwashits()</code></a></p></li>
<li><p><a href='#method-scan_rsid'><code>rms2$scan_rsid()</code></a></p></li>
<li><p><a href='#method-colocalise_rsid'><code>rms2$colocalise_rsid()</code></a></p></li>
<li><p><a href='#method-plot_coloc'><code>rms2$plot_coloc()</code></a></p></li>
<li><p><a href='#method-coloc_scan'><code>rms2$coloc_scan()</code></a></p></li>
<li><p><a href='#method-mr'><code>rms2$mr()</code></a></p></li>
<li><p><a href='#method-mvmr'><code>rms2$mvmr()</code></a></p></li>
<li><p><a href='#method-mvmr_lasso'><code>rms2$mvmr_lasso()</code></a></p></li>
<li><p><a href='#method-clone'><code>rms2$clone()</code></a></p></li>
</ul>
<p><hr>
<a id="method-new"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>new()</code></h3>
<p>Initialise the rms2 class for a particular outcome GWAS</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>rms2</span><span class='op'>$</span><span class='fu'>new</span><span class='op'>(</span><span class='va'>igd_id</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>igd_id</code></dt><dd><p>The OpenGWAS ID to use</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-extract_gwashits"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>extract_gwashits()</code></h3>
<p>Extract GWAS hits for igd_id</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>rms2</span><span class='op'>$</span><span class='fu'>extract_gwashits</span><span class='op'>(</span>
  igd_id <span class='op'>=</span> <span class='va'>self</span><span class='op'>$</span><span class='va'>igd_id</span>,
  pval <span class='op'>=</span> <span class='fl'>5e-08</span>,
  kb <span class='op'>=</span> <span class='fl'>10000</span>,
  r2 <span class='op'>=</span> <span class='fl'>0.001</span>,
  pop <span class='op'>=</span> <span class='st'>"EUR"</span>
<span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>igd_id</code></dt><dd><p>Default=self$igd_id</p></dd>

<dt><code>pval</code></dt><dd><p>Significance threshold for detecting GWAS hit default=5e-8</p></dd>

<dt><code>kb</code></dt><dd><p>Clumping parameter default=10000</p></dd>

<dt><code>r2</code></dt><dd><p>Clumping parameter default=0.001</p></dd>

<dt><code>pop</code></dt><dd><p>1000 genome super population to use for clumping default='EUR'</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-scan_rsid"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>scan_rsid()</code></h3>
<p>Search OpenGWAS for associations with a particular GWAS hit</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>rms2</span><span class='op'>$</span><span class='fu'>scan_rsid</span><span class='op'>(</span><span class='va'>rsid</span>, pval <span class='op'>=</span> <span class='fl'>5e-08</span>, idlist <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>rsid</code></dt><dd><p>RSID</p></dd>

<dt><code>pval</code></dt><dd><p>Significance threshold for candidate traits default=5e-8</p></dd>

<dt><code>idlist</code></dt><dd><p>Which traits to search. When NULL (default) will search all available traits</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-colocalise_rsid"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>colocalise_rsid()</code></h3>
<p>Perform colocalisation between outcome trait and a candidate trait for a particular locus</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>rms2</span><span class='op'>$</span><span class='fu'>colocalise_rsid</span><span class='op'>(</span><span class='va'>rsid</span>, <span class='va'>trait_id</span>, radius <span class='op'>=</span> <span class='fl'>50000</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>rsid</code></dt><dd><p>RSID</p></dd>

<dt><code>trait_id</code></dt><dd><p>trait_id</p></dd>

<dt><code>radius</code></dt><dd><p>Radius around which to perform analysis default=50000</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-plot_coloc"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>plot_coloc()</code></h3>
<p>Plot region for a colocalisation analysis</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>rms2</span><span class='op'>$</span><span class='fu'>plot_coloc</span><span class='op'>(</span><span class='va'>rsid</span>, <span class='va'>trait_id</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>rsid</code></dt><dd><p>rsid</p></dd>

<dt><code>trait_id</code></dt><dd><p>trait_id</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-coloc_scan"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>coloc_scan()</code></h3>
<p>Perform colocalisation systematically for all detected candidate traits for a given region</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>rms2</span><span class='op'>$</span><span class='fu'>coloc_scan</span><span class='op'>(</span><span class='va'>rsid</span>, radius <span class='op'>=</span> <span class='fl'>50000</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>rsid</code></dt><dd><p>rsid i.e. one of the outcome GWAS hits</p></dd>

<dt><code>radius</code></dt><dd><p>Radius around which to perform analysis default=50000</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-mr"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>mr()</code></h3>
<p>Perform MR analysis of all candidate traits for a particular rsid</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>rms2</span><span class='op'>$</span><span class='fu'>mr</span><span class='op'>(</span>
  <span class='va'>rsid</span>,
  steiger_filtering <span class='op'>=</span> <span class='cn'>TRUE</span>,
  exclude_rsid_region <span class='op'>=</span> <span class='cn'>TRUE</span>,
  radius <span class='op'>=</span> <span class='fl'>250000</span>,
  mrmethod <span class='op'>=</span> <span class='st'>"mr_ivw"</span>
<span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>rsid</code></dt><dd><p>GWAS hit from which candidate traits are obtained</p></dd>

<dt><code>steiger_filtering</code></dt><dd><p>Whether to perform steiger filtering default=TRUE</p></dd>

<dt><code>exclude_rsid_region</code></dt><dd><p>Whether to exclude the known GWAS hit default=TRUE</p></dd>

<dt><code>radius</code></dt><dd><p>Excluded region radius default=250000</p></dd>

<dt><code>mrmethod</code></dt><dd><p>MR method default="mr_ivw"</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-mvmr"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>mvmr()</code></h3>
<p>Perform multivariable MR for all or some of the candidate traits related to a particular region</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>rms2</span><span class='op'>$</span><span class='fu'>mvmr</span><span class='op'>(</span><span class='va'>rsid</span>, traitlist <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>rsid</code></dt><dd><p>RSID from which candidate traits are obtained</p></dd>

<dt><code>traitlist</code></dt><dd><p>List of traits to use. Default (NULL) is to use all, but important to scrutinise and manually exclude those which appear to be synonymous with the outcome</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-mvmr_lasso"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>mvmr_lasso()</code></h3>
<p>This function will take the output from mvmr, perform feature selection from the analysed traits using MVMR-LASSO, and then re-estimate the MVMR associations with only the retained traits</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>rms2</span><span class='op'>$</span><span class='fu'>mvmr_lasso</span><span class='op'>(</span><span class='va'>rsid</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>rsid</code></dt><dd><p>Region from which MVMR analysis has been performed</p></dd>

</dl><p></div></p>
<p><hr>
<a id="method-clone"></a></p><h3 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Method <code>clone()</code></h3>
<p>The objects of this class are cloneable with this method.</p><h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Usage</h4>
<p><div class="r"></p><pre><span class='va'>rms2</span><span class='op'>$</span><span class='fu'>clone</span><span class='op'>(</span>deep <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></pre><p></div></p>

<h4 class='hasAnchor' id='arguments'><a class='anchor' href='#arguments'></a>Arguments</h4>
<p><div class="arguments"></p><dl>
<dt><code>deep</code></dt><dd><p>Whether to make a deep clone.</p></dd>

</dl><p></div></p>



  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Gibran Hemani.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


